source/service.go

```diff
@@ -238,26 +238,28 @@ func (sc *serviceSource) extractHeadlessEndpoints(svc *v1.Service, hostname stri
 
 	targetsByHeadlessDomain := make(map[string][]string)
 	for _, v := range pods {
-		headlessDomain := hostname
+		headlessDomains := []string{hostname}
+
 		if v.Spec.Hostname != "" {
-			headlessDomain = v.Spec.Hostname + "." + headlessDomain
+			headlessDomains = append(headlessDomains, v.Spec.Hostname+"."+hostname)
```

Comment:
One question regarding this change, in the previous version `headlessDomain` gets overwritten if the hostname in spec in not empty. In your version you append it including headlessDomains := []string{hostname}. Is this correct?

Reply:
Yes, its correct because in case with statefulset any pod has stable hostname, saved in v.Spec.Hostname. Previous version created only 'v.Spec.Hostname + "." + headlessDomain' record in this case so we need to create 'headlessDomain' record  too